presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchSplitManager.java

```diff
@@ -44,27 +44,26 @@ public ElasticsearchSplitManager(ElasticsearchClient client)
     }
 
     @Override
-    public ConnectorSplitSource getSplits(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorTableLayoutHandle layout, SplitSchedulingStrategy splitSchedulingStrategy)
+    public ConnectorSplitSource getSplits(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorTableHandle table, SplitSchedulingStrategy splitSchedulingStrategy)
     {
-        ElasticsearchTableLayoutHandle layoutHandle = (ElasticsearchTableLayoutHandle) layout;
-        ElasticsearchTableHandle tableHandle = layoutHandle.getTable();
-        ElasticsearchTableDescription table = client.getTable(tableHandle.getSchemaName(), tableHandle.getTableName());
+        ElasticsearchTableHandle tableHandle = (ElasticsearchTableHandle) table;
+        ElasticsearchTableDescription tableDescription = client.getTable(tableHandle.getSchemaName(), tableHandle.getTableName());
         verify(table != null, "Table no longer exists: %s", tableHandle.toString());
 
-        List<String> indices = client.getIndices(table);
+        List<String> indices = client.getIndices(tableDescription);
         ImmutableList.Builder<ConnectorSplit> splits = ImmutableList.builder();
         for (String index : indices) {
-            ClusterSearchShardsResponse response = client.getSearchShards(index, table);
+            ClusterSearchShardsResponse response = client.getSearchShards(index, tableDescription);
             DiscoveryNode[] nodes = response.getNodes();
             for (ClusterSearchShardsGroup group : response.getGroups()) {
                 int nodeIndex = group.getShardId().getId() % nodes.length;
                 ElasticsearchSplit split = new ElasticsearchSplit(
                         index,
-                        table.getType(),
+                        tableDescription.getType(),
                         group.getShardId().getId(),
                         nodes[nodeIndex].getHostName(),
                         nodes[nodeIndex].getAddress().getPort(),
-                        layoutHandle.getTupleDomain());
+                        tableHandle.getConstraint());
```

Comment:
The table handle is now available in getRecordSet, so we should be able to create the recordset directly with the constraint extracted from the handle instead of passing it via the splits.